python/lsst/cell_coadds/_fits.py

```diff
@@ -0,0 +1,407 @@
+            outer=image_planes,
+            psf=psf,
+            inner_bbox=Box2I(
+                minimum=Point2I(
+                    header["GRCELL1"] * data["cell_id"][0],
+                    header["GRCELL2"] * data["cell_id"][1],
+                ),
+                maximum=Point2I(
+                    header["GRCELL1"] * (data["cell_id"][0] + 1) - 1,
+                    header["GRCELL2"] * (data["cell_id"][1] + 1) - 1,
+                ),
+            ),
+            common=common,
+            identifiers=identifiers,
+            inputs=None,  # type: ignore[arg-type]  # TODO: Pass a sensible value here.
+        )
+
+    # TODO: Make this reader a context manager that handles file closures.
+
+    def readBBox(self) -> Box2I:
+        raise NotImplementedError("Not yet implemented for MultipleCellCoadd")
+
+    def readPsf(self) -> ImageF:
+        raise NotImplementedError("Not yet implemented for MultipleCellCoadd")
+
+    def readFilter(self) -> str:
+        """Read the filter information from the FITS file.
+
+        Returns
+        -------
+        band : `str`
+            The filter information read from the FITS file.
+        """
+        with fits.open(self.filename) as hdu_list:
+            band = hdu_list[1].header["BAND"]
+        return band
+
+    def readXY0(self) -> Point2I:
+        raise NotImplementedError("Not yet implemented for MultipleCellCoadd")
+
+    def readWcs(self) -> afwGeom.SkyWcs:
+        """Read the WCS information from the FITS file.
+
+        Returns
+        -------
+        wcs : `~lsst.afw.geom.SkyWcs`
+            The WCS information read from the FITS file.
+        """
+        # Read in WCS
+        ps = PropertySet()
+        with fits.open(self.filename) as hdu_list:
+            ps.update(hdu_list[0].header)
+        wcs = afwGeom.makeSkyWcs(ps)
+        return wcs
+
+
+def writeMultipleCellCoaddAsFits(
+    multiple_cell_coadd: MultipleCellCoadd,
+    filename: str,
+    overwrite: bool = False,
+    metadata: PropertySet | None = None,
+) -> None:
+    """Write a MultipleCellCoadd object to a FITS file.
+
+    Parameters
+    ----------
+    multiple_cell_coadd : `MultipleCellCoadd`
+        The multiple cell coadd to write to a FITS file.
+    filename : `str`
+        The name of the file to write to.
+    overwrite : `bool`, optional
+        Whether to overwrite the file if it already exists?
+    """
+    array = [cell.identifiers.cell for cell in multiple_cell_coadd.cells.values()]
+    cell_id = fits.Column(name="cell_id", format="2I", array=array)
+
+    array = [cell.outer.image.array for cell in multiple_cell_coadd.cells.values()]
+    unit_array = [cell.common.units.name for cell in multiple_cell_coadd.cells.values()]
+    image = fits.Column(
+        name="image",
+        unit=unit_array[0],
+        format=f"{array[0].size}E",
+        dim=f"({array[0].shape[1]}, {array[0].shape[0]})",
+        array=array,
+    )  # make it a dask array for performance?
```

Comment:
You can drop this comment; I'm pretty sure `dask` would only help us if we gave it multiple threads, and I don't see us ever doing that.